src/picongpu/include/particles/ionization/byField/BSI/BSI.hpp

```diff
@@ -39,23 +50,157 @@ namespace ionization
     struct BSI
     {
 
-        /* define ionization ALGORITHM for ionization MODEL */
-        typedef particles::ionization::AlgorithmBSI IonizationAlgorithm;
-
-        /** Functor implementation
-         * \tparam T_SrcSpecies particle species to be ionized
-         * \tparam T_ParticleStorage \see picongpu/src/libPMacc/include/math/MapTuple.hpp
-         *
-         * \param currentStep current time step */
-        template<typename T_SrcSpecies,typename T_ParticleStorage>
-        void operator()(T_SrcSpecies& src, T_ParticleStorage& pst, const uint32_t currentStep)
-        {
-
-            /* alias for electron species to be created */
-            PMACC_AUTO(electrons,pst[typename MakeIdentifier<T_DestSpecies>::type()]);
-            /* call the ionization function */
-            src.ionize(electrons, currentStep);
-        }
+        private:
+
+            /* define ionization ALGORITHM (calculation) for ionization MODEL */
+            typedef particles::ionization::AlgorithmBSI IonizationAlgorithm;
+            /* stores blocks and block size for the kernel to start */
+            MappingDesc* cellDescription;
```

Comment:
Yes, therefore you should split your functor in two parts. One which is generic and know how to start a kernel with a Ionizer algorithm. 
The secound is the Ionizer (field caching, ionize). The Ionizer can handle all its dependencies like (store field-DataBoxes)